skopt/learning/gaussian_process/tests/test_kernels.py

```diff
@@ -44,11 +41,27 @@ def numerical_gradient(X, Y, kernel):
         grad.append(num_grad)
     return np.asarray(grad)
 
-def check_gradient_correctness(kernel):
+def check_gradient_correctness(kernel, X, Y):
     X_grad = kernel.gradient_x(X, Y)
     num_grad = numerical_gradient(X, Y, kernel)
     assert_array_almost_equal(X_grad, num_grad, decimal=3)
 
 def test_gradient_correctness():
+    rng = np.random.RandomState(0)
+    X = rng.randn(5)
+    Y = rng.randn(10, 5)
     for kernel in KERNELS:
-        yield check_gradient_correctness, kernel
+        yield check_gradient_correctness, kernel, X, Y
+
+def test_gradient_finiteness():
```

Comment:
Nods, I was thinking of: can we setup a small problem that leads to the old calculation producing infinite gradients but the new one doesn't do that. Does that make sense? (Not sure I am doing a good job explaining it)